core/station/impl/src/services/system.rs

```diff
@@ -356,30 +361,43 @@ mod install_canister_handlers {
     use crate::core::ic_cdk::api::{id as self_canister_id, print};
     use crate::core::init::{default_policies, DEFAULT_PERMISSIONS};
     use crate::core::INITIAL_UPGRADER_CYCLES;
-    use crate::models::{AddRequestPolicyOperationInput, EditPermissionOperationInput};
+    use crate::mappers::blockchain::BlockchainMapper;
+    use crate::mappers::HelperMapper;
+    use crate::models::permission::Allow;
+    use crate::models::request_specifier::UserSpecifier;
+    use crate::models::{
+        AddAccountOperationInput, AddRequestPolicyOperationInput, EditPermissionOperationInput,
+        RequestPolicyRule, ADMIN_GROUP_ID,
+    };
     use crate::services::permission::PERMISSION_SERVICE;
+    use crate::services::ACCOUNT_SERVICE;
     use crate::services::REQUEST_POLICY_SERVICE;
     use candid::{Encode, Principal};
     use canfund::fetch::cycles::FetchCyclesBalanceFromCanisterStatus;
     use canfund::manager::options::{EstimatedRuntime, FundManagerOptions, FundStrategy};
     use canfund::FundManager;
     use ic_cdk::api::management_canister::main::{self as mgmt};
-    use station_api::SystemInit;
+    use orbit_essentials::types::UUID;
+    use station_api::{InitAccountInput, SystemInit};
     use std::cell::RefCell;
     use std::sync::Arc;
 
     thread_local! {
         pub static FUND_MANAGER: RefCell<FundManager> = RefCell::new(FundManager::new());
     }
 
+    // Calculates the initial quorum based on the number of admins and the provided quorum, if not provided
+    // the quorum is set to the majority of the admins.
+    pub fn calc_initial_quorum(admin_count: u16, quorum: Option<u16>) -> u16 {
+        quorum.unwrap_or(admin_count / 2 + 1).clamp(1, admin_count)
+    }
+
     /// Registers the default configurations for the canister.
     pub async fn init_post_process(init: &SystemInit) -> Result<(), String> {
-        let admin_count = u16::try_from(init.admins.len()).unwrap_or(u16::MAX);
-
-        let admin_quorum = init
-            .quorum
-            .unwrap_or(admin_count / 2 + 1)
-            .clamp(1, admin_count);
+        let admin_quorum = self::calc_initial_quorum(
+            u16::try_from(init.admins.len()).unwrap_or(u16::MAX),
```

Comment:
Shouldn't we fail if there are more than u16::MAX admins rather than silently capping at u16::MAX?

Reply:
just added a fail early inside the init hook itself, this way we don't need to check this here.